### ISSUE TITLE
Rat, Wolf, Pig and Snake allied video messages

### ISSUE THREAD

**Issue by PitchNeeded** [Sh0wdown/UnofficialCrusaderPatch#21]
It would good to somehow provide Rat, Pig, Snake and Wolf with video messages if you are allied with him instead of 'my master informs you' all the time from their servant. It would make it more fun to be allied with them. Stronghold 1 is a good source of quotes from them, here are parts of their speech in Stronghold 1 that could be used....

Snake refuses to defend the player: 'Your men can take care of a few snivelling outlaws' : 0:50 https://www.youtube.com/watch?v=9w9Lmal1HNM

Snake agrees to defend the player:  'Show some backbone, my contingency plan is already in motion. Trust me' : 1:10 https://www.youtube.com/watch?v=g1V20nu7K_I

Snake requests help: 'Ah - I find myself in desperate need of reinforcements' 0:17 https://www.youtube.com/watch?v=CZRVibFVMQ4

Snake agrees to send goods: 'any conditions you make will be honoured' 1;27:58 https://www.youtube.com/watch?v=J6YUvYraP6Y

Rat refuses to attack an enemy: 'I still don't have enough troops so' 1:25 https://www.youtube.com/watch?v=W3XuQhJtGvc

Rat refuses to defend the player 'My troops have scattered all over the place, there is nothing I can do' 1:02 https://www.youtube.com/watch?v=g1V20nu7K_I

Pig agrees to defend the player: 'I'll go break some heads' 1:12  https://www.youtube.com/watch?v=fUZw-Tf-exU

Pig agrees to attack an enemy: 'Some of my men are nearby, I will send them along to join in the fun hahaha' 0:49 https://www.youtube.com/watch?v=CZRVibFVMQ4

Wolf refuses to defend the player 'This is a waste of my time' 1:21 https://www.youtube.com/watch?v=QPsXI-IaQl8

Wolf agrees to attack: 'I need to deal with the source of the problem' 0:59 https://www.youtube.com/watch?v=JRa6n9n4Knc

Wolf agrees to defend the player 'My main force is marching half the length of the country to get to you' 0:35 https://www.youtube.com/watch?v=CZRVibFVMQ4

Wolf refuses to send goods: 'Silence!' 0:50 https://www.youtube.com/watch?v=JRa6n9n4Knc

(an allied) Wolf reacts to you killing an enemy: 'Impressive' 31:32 https://www.youtube.com/watch?v=XfjMKOWuXjk

**Comment by Sh0wdown** [comment]
This is a nice idea but needs files from Stronghold 1. Without FireFly's blessing to use those files, I won't look into this.

**Comment by Evrey** [comment]
You could instead ask the user to provide an installation directory of SH1. *(Do **not** just guess it or default to Steam or whatever.)* Then it's all just a matter of "What language is this?" and then - based on the language - auto-cut and copy the necessary audio files.

**Comment by PitchNeeded** [comment]
> You could instead ask the user to provide an installation directory of SH1. (Do not just guess it or default to Steam or whatever.) Then it's all just a matter of "What language is this?" and then - based on the language - auto-cut and copy the necessary audio files.

Yeah you should make it so that you **have to have** Stronghold 1 to use this feature (a screen comes up when installing the patch and asks you to provide the directory for Stronghold 1, and checks that the full version of Stronghold 1 is there). Some of the quotes I suggested would be just part of the audio files though, so autocutting wouldn't work.

**Comment by Evrey** [comment]
> and checks that the full version of Stronghold 1 is there
Don't. Just check whether the expected audio files exist in the expected directories or not. Perhaps guess the language by also reading the compiled ingame text file. Btw., cutting audio means taking part of a longer audio stream. Has nothing to do with Cut-and-Paste.

**Comment by PitchNeeded** [comment]
Yeah, it would make it more fun to ally with Wolf, Pig, Snake or Rat.

Also considering the excellent progress Sh0wdown and others are making is making with changing AI programming, that ugly servant who talks about his 'master' that comes up at moment when you are allied with Rat, Pig, Snake or Wolf could be turned into a new evil AI character one day..... 😃

**Comment by ByBurton** [comment]
> could be turned into a new evil AI character one day..... 😃

Please no. I would not like that. I would prefer to have "Robin Hood" instead.
An AI that builds a ton of archers, and very few spearmen. Attacks with no siege engines and only archers.
Spearmen are only for defense.
Attacks in huge numbers, and uses - like The Snake - archers to attack walls as well.

<3

**Comment by Lolasik011** [comment]
I would suggest Lord Woolsack because he already has a set of videos and sounds.

**Comment by PitchNeeded** [comment]
> I would suggest Lord Woolsack because he already has a set of videos and sounds and icons. I think this is a very good choice.

A new AI would be hard to do, the best thing to do would be use the template of an existing AI's coding to create a new AI. So to create my proposed 'Lord Woolsack', the template of Richard could be used so he would have exactly the same economy as him, but it would have to be changed his AI behaviour (if it is possible) regarding his military characteristics (not recruiting swordsmen but more pikemen instead, more pikemen sent on raids than Richard does, monks being trained if he has a cathedral etc).

**Comment by Flechador** [comment]
> Please no. I would not like that. I would prefer to have "Robin Hood" instead.
> An AI that builds a ton of archers, and very few spearmen. Attacks with no siege engines and only archers.
> Spearmen are only for defense.
> Attacks in huge numbers, and uses - like The Snake - archers to attack walls as well.

Robin Hood would then use exactly the same units as the rat would and that makes him to simmilar to him. Of course many archers makes sence with an roobin hood AI but he could use the classic Bandit unit, the maceman, insteat of the spearmen. And some shields would make sence if you as me.
I dont understand why it should be good if the archers attack the walls, thats extremly inefficient.

**Comment by Lolasik011** [comment]
Yes, but for "Robin Hood" it is necessary to draw new versions of the video and make audio voice  and icons. Of course, you could use the audio and video advisor for Robin Hood. Yes, and I would like not to replace some kind of AI and add  new AI. I would like to add Lord Woolsack.

By the way, has anyone seen the cut video of the attack Pigs from the demo (Where he eats from a bowl
 :D)

**Comment by PitchNeeded** [comment]
> > Please no. I would not like that. I would prefer to have "Robin Hood" instead.
> > An AI that builds a ton of archers, and very few spearmen. Attacks with no siege engines and only archers.
> > Spearmen are only for defense.
> > Attacks in huge numbers, and uses - like The Snake - archers to attack walls as well.
> 
> Robin Hood would then use exactly the same units as the rat would and that makes him to simmilar to him. Of course many archers makes sence with an roobin hood AI but he could use the classic Bandit unit, the maceman, insteat of the spearmen. And some shields would make sence if you as me.
> I dont understand why it should be good if the archers attack the walls, thats extremly inefficient.

But if Robin Hood used as melee troops tunnelers instead of spearmen that would make him more unique :smile: Tunnelers are better than spearmen I think: faster and they fight decently too.

**Comment by Flechador** [comment]
> But if Robin Hood used as melee troops tunnelers instead of spearmen that would make him more unique 😄 Tunnelers are better than spearmen I think: faster and they fight decently too.

Oh, never thought about using tunnelers as main troops, thats an pretty cool idea. 
Would love to see them more often. Because, right now only Richard is using them as support.

**Comment by GrosserBauerr2** [comment]
Found the Hex-Code for "bad_soldier_nevous.bik" it`s   90 25 5A 00 if you change it you can split the video message to the rat, snake, pig and wolf, for example rt_plead1.bik has the code  C8 27 5A 00 if you insert this code at the right one then the rat movie will apers when he asks for help.

**Comment by GrosserBauerr2** [comment]
I added a list with all Hex-bik-numbers from the rat, snake, pig, wolf and the bad soldier.
https://github.com/GrosserBauerr2/Stronghold-Crusader-AI
I also know now that the number  bad_soldier_nevous.bik 90 25 5A 00 is a referencenumber where the data of bad_soldier_nevous.bik (62 61 64 5F 73 6F 6C 64 69 65 72 5F 6E 65 76 6F 75 73 2E 62 69 6B 00) stays in the hex code.  For this you have split the number up in 90 25 turn it around and put 1A infront of it -> 1A 90 25 and this is the position of the data in the code. I think 5A 00 means that it is a message from the AI (it shows up in all AI message bik-  and wav-references) but not entiretly sure right now. I don t know how but i hope with this that it is possible to add complete new sound and bik files to the code.

**Comment by Xebenkeck** [comment]
There are lots of UNUSED fx audio files in the game already for The Rat, Snake, Pig and Wolf. We could use all the unused ones and swap around some of the existing ones to replace some of the lines from the "bad soldier."

I created a spreadsheet of some proposed voice line changes. I will post it when I get home tonight.

We would have to do some remapping in the .exe to call the right audio files and .bik' files so the proper ones appear, rather then the soldier. It looks like GrosserBauerr2 has found most of the .biks in the hex already.

And we would have to change the in-game text, which is simple to do with the text editor firefly gave us.

**Comment by KOYK** [comment]
> This is a nice idea but needs files from Stronghold 1. Without FireFly's blessing to use those files, I won't look into this.

I think you dont need to worry about it. also check their official discord, you will find this message under stronghold 1. 

https://i.imgur.com/uk4dc5O.png

**Comment by GrosserBauerr2** [comment]
Hello guys I don t know if here is still in interest in but I am working on a tool that splits up the messages. Right know it is able to split the movies from the bad soldier in different movies from the 4 lords. But now I need to add new sound files to the code and I am not able to do it with my own programming skills. It would be also nice to add some of the unused bik files like rt_vict2.bik or sn_plead1.bik because like sn_plead1 looks perfect for asking the snake for goods.
There are 12 Bad Soldier Messages so in total there are needed 48 new soundfiles to continue.

**Comment by Lolasik011** [comment]
> Hello guys I don t know if here is still in interest in but I am working on a tool that splits up the messages. Right know it is able to split the movies from the bad soldier in different movies from the 4 lords. But now I need to add new sound files to the code and I am not able to do it with my own programming skills. It would be also nice to add some of the unused bik files like rt_vict2.bik or sn_plead1.bik because like sn_plead1 looks perfect for asking the snake for goods.
> There are 12 Bad Soldier Messages so in total there are needed 48 new soundfiles to continue.

I think at Discord you'll get help

**Comment by LordHansCapon** [comment]
While programmatically this became solvable, we still dont have enough sound files to satisfy every aspect sadly.

**Comment by Xebenkeck** [comment]
There are 8 in the game already unused for the four lords,

Stronghold 1 doesn't have ones we could use? Based on that post from firefly we should be able to use them for audio.
